torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp

```diff
@@ -211,7 +248,10 @@ void insertQuantizationOps(
   }
   Value* original_val = observer->input(1);
   Node *quant, *choose_qparams, *dequant;
-  if (quant_type == QuantType::DYNAMIC && !isWeight(module, observer_out)) {
+  if (quant_type == QuantType::DYNAMIC && isNoopObserver(observer->input(0))) {
+    dequant = insertFP16CastOps(g, observer_out);
+  } else if (
+      quant_type == QuantType::DYNAMIC && !isWeight(module, observer_out)) {
```

Comment:
I feel it's more clear to merge the two branches into one? since the else branch is only for static I think.
if (quant_type == quanttype::dynamic) {
if (isNooposerver) {
} else if (!isWeight(..) {
}

} else {
...
}